File: MyApplication/app/src/main/java/com/homework/myapplication/adapter/QuestionAdapter.java
package com.homework.myapplication.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.homework.myapplication.R;
import com.homework.myapplication.bean.Exam;

import java.util.ArrayList;
import java.util.List;


public class QuestionAdapter extends RecyclerView.Adapter<QuestionAdapter.AnswerViewHolder> {
    private Context context;
    private List<Exam> examList = new ArrayList<>();
    private OnItemClickListener onItemClickListener;

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    public interface OnItemClickListener{
        void onItemClick(View v,int postion);
    }

    public QuestionAdapter(Context context, List<Exam> examList) {
        this.context = context;
        this.examList = examList;
    }

    @NonNull
    @Override
    public AnswerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_question,parent,false);
        return new AnswerViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull AnswerViewHolder holder, int position) {
        Exam exam = examList.get(position);
        //holder.tvQuestion.setText(exam.get);
        holder.tvQuestion.setText("Question: "+exam.getQuestionContent());
        holder.tvIscorrect.setText(exam.getIsCorrect()==1?"true":"false");
    }

    @Override
    public int getItemCount() {
        return examList.size();
    }

    public class AnswerViewHolder extends RecyclerView.ViewHolder {
        private TextView tvIscorrect;
        private TextView tvQuestion;

        public AnswerViewHolder(@NonNull View itemView) {
            super(itemView);
            tvIscorrect = itemView.findViewById(R.id.tv_iscorrect);
            tvQuestion = itemView.findViewById(R.id.tv_question);
        }

    }
}

File: MyApplication/app/src/main/java/com/homework/myapplication/adapter/QuestionAdapter.kt
package com.homework.myapplication.adapter

import android.app.Application
import com.homework.myapplication.model.DaoManager
import com.homework.myapplication.interfaces.CallBack
import com.homework.myapplication.bean.Score
import android.app.Activity
import android.content.Intent
import com.homework.myapplication.controller.ResultActivity
import com.homework.myapplication.app.MyApplication
import com.homework.myapplication.bean.Exam
import com.homework.myapplication.bean.Answer
import com.homework.myapplication.util.SharePerferenceUtils
import com.homework.myapplication.util.DataUtil
import com.homework.myapplication.bean.Question
import com.homework.myapplication.util.ListUtil
import android.database.sqlite.SQLiteOpenHelper
import com.homework.myapplication.util.DataBaseHelper
import android.database.sqlite.SQLiteDatabase
import android.content.SharedPreferences
import android.content.ContentValues
import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import com.homework.myapplication.R
import android.text.TextUtils
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.widget.TextView
import com.homework.myapplication.adapter.ResultAdapter.ResultHolder
import androidx.recyclerview.widget.LinearLayoutManager
import com.homework.myapplication.adapter.QuestionAdapter
import androidx.fragment.app.FragmentPagerAdapter
import android.os.Bundle
import kotlin.jvm.JvmOverloads
import androidx.appcompat.app.AppCompatActivity
import com.homework.myapplication.controller.NoScrollViewPager
import com.homework.myapplication.util.GuidGeneratorUtil
import com.homework.myapplication.controller.ExamFragment
import com.homework.myapplication.adapter.MyFragmentPagerAdapter
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import android.widget.Toast
import com.homework.myapplication.util.PecentUtil
import android.os.Build
import com.homework.myapplication.adapter.OptionAdapter
import android.widget.EditText
import com.homework.myapplication.controller.ExamActivity
import com.homework.myapplication.adapter.ResultAdapter
import androidx.viewpager.widget.ViewPager
import android.view.MotionEvent
import android.view.View
import java.util.ArrayList

class QuestionAdapter(private val context: Context, examList: List<Exam?>?) :
    RecyclerView.Adapter<QuestionAdapter.AnswerViewHolder>() {
    private val examList: List<Exam?>? = ArrayList()
    private var onItemClickListener: OnItemClickListener? = null
    fun setOnItemClickListener(onItemClickListener: OnItemClickListener?) {
        this.onItemClickListener = onItemClickListener
    }

    interface OnItemClickListener {
        fun onItemClick(v: View?, postion: Int)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AnswerViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_question, parent, false)
        return AnswerViewHolder(view)
    }

    override fun onBindViewHolder(holder: AnswerViewHolder, position: Int) {
        val exam = examList!![position]
        //holder.tvQuestion.setText(exam.get);
        holder.tvQuestion.text = "Question: " + exam.getQuestionContent()
        holder.tvIscorrect.text = if (exam.getIsCorrect() == 1) "true" else "false"
    }

    override fun getItemCount(): Int {
        return examList!!.size
    }

    inner class AnswerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvIscorrect: TextView
        val tvQuestion: TextView

        init {
            tvIscorrect = itemView.findViewById(R.id.tv_iscorrect)
            tvQuestion = itemView.findViewById(R.id.tv_question)
        }
    }

    init {
        this.examList = examList
    }
}